core/src/main/scala/org/apache/spark/scheduler/StageInfo.scala

```diff
@@ -107,4 +107,7 @@ private[spark] object StageInfo {
       taskLocalityPreferences,
       shuffleDepId)
   }
+
+  val INVALID_STAGE_ID = -1
```

Comment:
`val` -> `final val`.

Reply:
Thanks.
I've noticed there are too much `final`-less `val` declared in objects as a constant value in the code base...
Exactly, `val` in object should not be overridden but it's not marked as `final` in byte code level so I'll put `final` modifier just in case.